### DISCUSSION TITLE
a unique question blocking client(s)

### DISCUSSION THREAD

**Discussion by allhailCdosdude** [AdguardTeam/AdGuardHome#3786] [Q&A]
What I want to do is block certain clients from access to the Internet, except for some Internet sites. Confusing, I know. So, I restate: a client can have access to the internet, but ONLY to a few or one website(s). Is this possible? 

A second question, is it possible to limit access by time of day?

Thanks a lot.

**Comment by ainar-g** [comment]
>  a client can have access to the internet, but ONLY to a few or one website(s).

This can be achieved using a combination of custom rules.  First, block everything.  Then, allow some sites.  For example:

```none
/.*/$client=ClientName
@@||good.example.com^$client=ClientName
```

 >  A second question, is it possible to limit access by time of day?

Not currently.  See #951.

**Reply by allhailCdosdude** [reply]
> > a client can have access to the internet, but ONLY to a few or one website(s).
> 
> This can be achieved using a combination of custom rules. First, block everything. Then, allow some sites. For example:
> 
> ```
> /.*/$client=ClientName
> @@||good.example.com^$client=ClientName
> ```
> 
> > A second question, is it possible to limit access by time of day?
> 
> Not currently. See #951.

Thanks a lot. To test, I put this in "Custom filtering rules"
/.*/$client=autumn-a.lan
@@||https://www.ebay-kleinanzeigen.de^$client=autumn-a.lan

And then applied. 
I got positive/green update, bottom right of the screen.
I ran check and got no errors.
But when I went to bbc.com, I was able to get through.

Did I do it right? And are there any other steps necessary after hitting 'apply.'

Many thanks.

**Reply by ainar-g** [reply]
Please read the [documentation].  `client=` is supposed to be the client's name, not domain.

[documentation]: https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#client

**Comment by allhailCdosdude** [comment]
That is not a domain, though it looks like it. That is the name of a laptop on the network. Can I use the client's IP instead?

**Reply by ainar-g** [reply]
Yes, as is written in the [documentation].

[documentation]: https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#client

**Comment by sddzcuigc** [comment]
I have a question , I need to block someclients without name,because they join my dns without my accepttion , they use the doh without any client name, https:// my dns name/dns-query/    how can I block them?
